src/main/java/org/nuxeo/ecm/core/storage/marklogic/MarkLogicRepository.java

```diff
@@ -672,13 +682,21 @@ private State findOne(String ctsQuery) {
             logQuery(query);
         }
         // Run query
-        try (Session session = xccContentSource.newSession()) {
-            AdhocQuery request = session.newAdhocQuery(query);
-            // ResultSequence will be closed by Session close
+        try {
+            // Don't auto close session it is the stream responsibility
+            Session session = xccContentSource.newSession();
```

Comment:
But the idea is to be robust for any kind of exceptions, even unchecked ones like NPE or OutOfMemory etc. So as soon as you allocate a session, there must be a try/finally to close it if it's not returned in a wrapper that itself takes responsibility for the closing.